### PR TITLE
Version 4

### LINKED ISSUE #115: palette colours org-b2b-* to be deprecated?
Unless they are being used by anyone else KAT was the only product that used org-b2b-* colours

### PR DESCRIPTION
**Pull request by onishiweb**
This version introduces:
- Brand new color palette with just 21 colors!
- Support for generated text colors in `oColorsFor`
- New function for tinting palette colors: `oColorsGetTint`
- New function to generate text colors: `oColorsGetTextColor`
- New function to mix two colors: `oColorsMix`
- Support for CSS Variables (through non-silent mode and the build service)
- New accessibility functions to help text color combinations when generating text colors
- New use cases for opinion, hero, and highlight brands

Closes #115 #118

### CODE REVIEW

**Comment by onishiweb** [comment]
The demos are still to be completed so there may be some outstanding issues there.

**Comment by onishiweb** [comment]
Please do not merge, but do review this PR :)

**Comment by onishiweb** [comment]
Ready to review!

File: README.md [hunk 1]
```diff
@@ -1,27 +1,71 @@
-# Colours [![Build Status](https://circleci.com/gh/Financial-Times/o-colors.png?style=shield&circle-token=0d3884dacba2fdd8aec8afbd2eaa84c79040ddd5)](https://circleci.com/gh/Financial-Times/o-colors)
+o-colors [![Build Status](https://circleci.com/gh/Financial-Times/o-colors.png?style=shield&circle-token=0d3884dacba2fdd8aec8afbd2eaa84c79040ddd5)](https://circleci.com/gh/Financial-Times/o-colors)
```
Comment (rowanmanning) [inline README.md]: Any reason why the `#` was removed?

File: README.md [hunk 2]
```diff
@@ -1,27 +1,71 @@
...
+```scss
+$o-colors-is-silent: false;
+@import 'o-colors/main';
+```
+
+#### Colours and accessibility
+
+o-colors has been built to help bridge the gap between design and development by providing functionality to help create colours dynamically from a central palette as well as generate contrasted text colours based on an elements background color.
+
+You can create tints of a color with the [oColorsGetTint](#tint-palette-colors) function. The function takes a palette color name and a brightness value (based on HSB color) to return a tint of the palette color.
```
Comment (rowanmanning) [inline README.md]: Would maybe backtick these so that they appear clearly as method names when scanning

Comment (onishiweb) [inline README.md]: 👍

File: README.md (hunk 1 above)
Comment (onishiweb) [inline README.md]: by accident I think

File: README.md [hunk 3]
```diff
@@ -1,27 +1,71 @@
...
+	- [Upgrading from v3.x.x to v4.x.x](#upgrading-from-v3.x.x-to-v4.x.x)
+- [Contact](#contact)
+- [Licence](#licence)
 
-This is an [Origami](http://origami.ft.com/) module that provides variables defining the FT digital colour palette, and helpers to use them in your products.  For installation instructions, see the [registry page](http://registry.origami.ft.com/components/o-colors).
 
 ## Usage
 
-There are a number of ways of using colours in your component or product.  Here they are from most to least preferred.  All mixin and function approaches require you to import the module into your own build, while the final method of using predefined classes is compatible with using the build service.
+There are a number of ways of using colors in your component or product. o-colors can be used via the [Build Service](https://www.ft.com/__origami/service/build/v2/), but it is recommended you import the Sass into your project to make use of the many Sass mixins and functions available.
```
Comment (JakeChampion) [inline README.md]: Should we document how to import the Sass into a project?

Comment (onishiweb) [inline README.md]: It's in the Sass block below :)

File: demos/src/demo.scss [hunk 4]
```diff
@@ -1,80 +1,241 @@
...
+	@return $json;
 }
-.demo-descrip {
-	font-family: monospace;
-	margin-left: 10px;
-	font-size: 75%;
-	color: #777777;
+
+body {
+	content: "#{oColorsJSONifyMap($o-colors-palette)}";
```
Comment (JakeChampion) [inline demos/src/demo.scss]: Very clever!

File: demos/src/demo.scss [hunk 5]
```diff
@@ -1,80 +1,241 @@
...
+	font-family: Helvetica, Arial, sans-serif;
+
+	*,
+	*:before,
+	*:after {
+		box-sizing: inherit;
+	}
 }
 
+@debug "Palette: #{$o-colors-palette}";
```
Comment (JakeChampion) [inline demos/src/demo.scss]: Do we need the debug statement?

File: demos/src/demo.scss [hunk 6]
```diff
@@ -1,80 +1,241 @@
...
+@each $name, $csscolor in $o-colors-palette {
+	@if $csscolor != 'transparent' and $csscolor != 'inherit' {
+		.o-colors-palette-#{$name} .hex,
+		.o-colors-palette-#{$name}:after {
+			@if $name == 'jade' {
```
Comment (JakeChampion) [inline demos/src/demo.scss]: How come jade is a special case, could we have a comment about this?

File: demos/src/demo.scss [hunk 7]
```diff
@@ -1,80 +1,241 @@
...
+	margin: 0 auto;
+}
+
+.demo-wrapper {
+	width: 100%;
+	display: flex;
+	margin-left: -10px;
+	margin-right: -10px;
+	justify-content: flex-start;
+	flex-direction: row;
```
Comment (JakeChampion) [inline demos/src/demo.scss]: Can demos use other Origami components? If so, should we use o-grid and o-buttons?

File: demos/src/palette.json [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+			},
+			{
+				"name": "teal",
+				"tints": true,
+				"usage": "text, backgrounds, buttons, links"
+			}
+		]
+	},
+	"secondary" : {
+		"description": "Hello world this is a group description",
```
Comment (JakeChampion) [inline demos/src/palette.json]: Template description should be changed.

File: demos/src/palette.json [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+				"usage": "backgrounds, text"
+			},
+			{
+				"name": "lemon",
+				"usage": "backgrounds, text"
+			}
+		]
+	},
+	"tertiary" : {
+		"description": "Hello world this is a group description",
```
Comment (JakeChampion) [inline demos/src/palette.json]: Template description should be changed.

File: demos/src/demo.scss (hunk 7 above)
Comment (onishiweb) [inline demos/src/demo.scss]: Possibly, but this was a little custom so I _think_ it's ok? Happy to go through it after release and fix it all up.

File: demos/src/demo.scss (hunk 4 above)
Comment (onishiweb) [inline demos/src/demo.scss]: Why thank you :)

File: README.md [hunk 10]
```diff
@@ -80,32 +177,35 @@ It takes three arguments:
...
-Instructions for maintaining the color palette for non-developers:
+### Upgrading from v3.x.x to v4.x.x
 
-### Getting started
+o-colors v4.x.x updates the entire palette of colors and adds a lot more functionality through new mixins and functions. The palette has been reduced from over 60 colors to a base palette of around 20 colors that can be manipulated through the new functionality in v4.x.x.
+
+To migrate from v3.x.x to use v4.x.x you will need to update the palette colors you are requesting using `oColorsFor`, `oColorsSetUseCase`, and `oColorsGetPaletteColor`. To work out which color names you need to update, we've created a [table showing which colors should now be used in place of the old v3.x.x palette colors](migration-guide.md).
+
+The use cases have also been updated to add new use cases for opinion, hero, and highlight branding, but also removes support for the `product-brand` use case.
+
```
Comment (alicebartlett) [inline README.md]: [restructure suggestion]
New use cases have been added for `opinion`, `hero` and `highlight` branding.
The `product-brand` use case has been removed.

File: demos/src/demo.js [hunk 11]
```diff
@@ -1,81 +1,157 @@
...
+		let oColor = swatch.getAttribute('data-o-color');
+		let hexInput = swatch.querySelector('.hex');
+		hexInput.value = palette[oColor];
+
+		if (clickToCopy) {
+			swatch.addEventListener('click', oColorsCopy, false);
+		}
 	}
+
+	// populateTintDemos(palette, colorTints);
```
Comment (alicebartlett) [inline demos/src/demo.js]: commented out line here?

File: demos/src/demo.js [hunk 12]
```diff
@@ -1,81 +1,157 @@
...
+		}
+	};
+}
+
+function oColorsShowUseCases() {
+	const useCase = this.textContent;
+	const useCaseClass = '.demo-color-for-' + useCase;
+
+	const colors = document.querySelectorAll(useCaseClass);
+	// const activeUseCases = document.querySelectorAll('.use-case-active');
```
Comment (alicebartlett) [inline demos/src/demo.js]: and here

File: src/scss/_functions.scss [hunk 13]
```diff
@@ -15,6 +15,27 @@
 	}
 }
 
+/// Returns a colour based on the background context and base color
```
Comment (alicebartlett) [inline src/scss/_functions.scss]: you've used colour and color in the same sentence here ;-p

**Pull request merged**